cmd/server.go

```diff
@@ -113,38 +113,42 @@ and processes log data according to the OpenTelemetry protocol.`,
 				switch sig {
 				case syscall.SIGTERM, syscall.SIGINT:
 					slog.Info("Received shutdown signal", "signal", sig)
-					pool.Close()
+
+					shutdownCtx, shutdownCancel := context.WithTimeout(context.Background(), shutdownTimeout)
+					defer shutdownCancel()
+
+					shutdownDone := make(chan struct{})
+					go func() {
+						defer close(shutdownDone)
+						srv.Stop()
+						httpSrv.Shutdown(shutdownCtx)
+						pool.Close()
+					}()
+
+					select {
+					case <-shutdownDone:
+						slog.Info("Server stopped gracefully")
+					case <-shutdownCtx.Done():
+						slog.Warn("Server shutdown timed out, forcing stop")
+						srv.ForceStop()
+					}
+
 					cancel()
+					return fmt.Errorf("shutdown signal received")
 				case syscall.SIGHUP:
 					slog.Info("Received reload signal", "signal", sig)
-					// TODO: Implement configuration reload
 				}
 			}
 			return nil
 		})
 
 		if err := g.Wait(); err != nil {
+			if err.Error() == "shutdown signal received" {
```

Comment:
I get `http: Server closed` which leads to printing out the command line help here

Reply:
same here 

<img width="778" height="289" alt="Image" src="https://github.com/user-attachments/assets/66ae3c31-9edd-495b-8075-193cc1e96601" />